src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreWorker.cs

```diff
@@ -322,15 +365,19 @@ public void CleanCache()
                     catch (OperationCanceledException) when (token.IsCancellationRequested)
                     {
                         // Ignores
+                        Logger.LogInformation($"Restore operation {restoreOperation} has been cancelled");
                     }
                     catch (Exception e)
                     {
                         // Writes stack to activity log
-                        _logger.Value.LogError(e.ToString());
+                        Logger.LogError(e.ToString());
                         // Do not die just yet
                     }
                 }
             }
+
+            // returns false on preliminary exit (cancellation)
+            return false;
```

Comment:
You might also throw `OperationCanceledException` from this method on cancellation. Are callers prepared to deal with both outcomes?